packages/contracts/contracts/BorrowerOperations.sol

```diff
@@ -332,17 +333,20 @@ contract BorrowerOperations is LiquityBase, Ownable, CheckContract, IBorrowerOpe
 
         _requireTroveisActive(troveManagerCached, msg.sender);
         uint256 price = priceFeed.fetchPrice();
-        _requireNotInRecoveryMode(price);
+        if (thusdTokenCached.mintList(address(this))) {
+            _requireNotInRecoveryMode(price);
+        }
 
         troveManagerCached.applyPendingRewards(msg.sender);
 
         uint256 coll = troveManagerCached.getTroveColl(msg.sender);
         uint256 debt = troveManagerCached.getTroveDebt(msg.sender);
 
         _requireSufficientTHUSDBalance(thusdTokenCached, msg.sender, debt - THUSD_GAS_COMPENSATION);
-
-        uint256 newTCR = _getNewTCRFromTroveChange(coll, false, debt, false, price);
-        _requireNewTCRisAboveCCR(newTCR);
+        if (thusdTokenCached.mintList(address(this))) {
```

Comment:
this line and line 336 will lead to 2 intercontract calls which is not cheap, but value can't be changed between them. How about to cache value from first call?